tests/symengine/symengine_tensor_operations_04.cc

```diff
@@ -161,7 +169,6 @@ main()
   test_tensor<4, dim>();
 
   test_symmetric_tensor<2, dim>();
-  test_symmetric_tensor<4, dim>();
```

Comment:
It would be nice if we could preserve this check. What's the reason why you're removing it?

Reply:
I know, but `unrolled_to_component_indices()` is not implemented for rank 4. Given the fact that we fill all tensors homogeneously with the same number in this test I simply opted to remove the rank 4 variant.

Reply:
I don't feel strongly about the issue, but I think that the function ought to exist:
```
template <int rank_, int dim, typename Number>
constexpr inline TableIndices<rank_>
Tensor<rank_, dim, Number>::unrolled_to_component_indices(const unsigned int i)
{
  // Work-around nvcc warning
  unsigned int dummy = n_independent_components;
  AssertIndexRange(i, dummy);
  (void)dummy;

  TableIndices<rank_> indices;

  unsigned int remainder = i;
  for (int r = rank_ - 1; r >= 0; --r)
    {
      indices[r] = internal::mod<dim>(remainder);
      remainder  = internal::div<dim>(remainder);
    }
  Assert(remainder == 0, ExcInternalError());

  return indices;
}
```

Reply:
@bangerth I would have not removed the test if it wouldn't trigger an exception. The problem is that for the `SymmetricTensor` (not `Tensor`) the function is not implemented: https://github.com/dealii/dealii/blob/d3f0a4a4d62b505188c3839b2434a66692f2243a/include/deal.II/base/symmetric_tensor.h#L2563-L2652